app/selectors/autocomplete.js

```diff
@@ -0,0 +1,198 @@
+            const match = value.match(regex);
+            lastValue = value;
+            lastIsSearch = isSearch;
+            if (match) {
+                lastMatchTerm = isSearch ? match[1] : match[2];
+            } else {
+                lastMatchTerm = null;
+            }
+        }
+        return lastMatchTerm;
+    };
+})();
+
+export const getMatchTermForChannelMention = (() => {
+    let lastMatchTerm = null;
+    let lastValue;
+    let lastIsSearch;
+    return (value, isSearch) => {
+        if (value !== lastValue || isSearch !== lastIsSearch) {
+            const regex = isSearch ? Autocomplete.CHANNEL_MENTION_SEARCH_REGEX : Autocomplete.CHANNEL_MENTION_REGEX;
+            const match = value.match(regex);
+            lastValue = value;
+            lastIsSearch = isSearch;
+            if (match) {
+                lastMatchTerm = isSearch ? match[1] : match[2];
+            } else {
+                lastMatchTerm = null;
+            }
+        }
+        return lastMatchTerm;
+    };
+})();
+
+export const filterMembersInChannel = createSelector(
+    getProfilesInCurrentChannel,
+    getCurrentUserId,
+    (state, matchTerm) => matchTerm,
+    (profilesInChannel, currentUserId, matchTerm) => {
+        let profiles;
+        if (matchTerm) {
+            profiles = profilesInChannel.filter((p) => {
+                return ((p.id !== currentUserId) && (
+                    p.username.toLowerCase().includes(matchTerm) || p.email.toLowerCase().includes(matchTerm) ||
+                    p.first_name.toLowerCase().includes(matchTerm) || p.last_name.toLowerCase().includes(matchTerm)));
+            });
+        } else {
+            profiles = profilesInChannel.filter((p) => p.id !== currentUserId);
+        }
+
+        // already sorted
+        return profiles.map((p) => p.id);
+    }
+);
+
+export const filterMembersNotInChannel = createSelector(
+    getProfilesNotInCurrentChannel,
+    getCurrentUserId,
+    (state, matchTerm) => matchTerm,
+    (profilesNotInChannel, currentUserId, matchTerm) => {
+        let profiles;
+        if (matchTerm) {
+            profiles = profilesNotInChannel.filter((p) => {
+                return ((p.id !== currentUserId) && (
+                    p.username.toLowerCase().includes(matchTerm) || p.email.toLowerCase().includes(matchTerm) ||
+                    p.first_name.toLowerCase().includes(matchTerm) || p.last_name.toLowerCase().includes(matchTerm)));
+            });
+        } else {
+            profiles = profilesNotInChannel;
+        }
+
+        return profiles.map((p) => p.id);
+    }
+);
+
+export const filterMembersInCurrentTeam = createSelector(
+    getProfilesInCurrentTeam,
+    getCurrentUser,
+    (state, matchTerm) => matchTerm,
+    (profilesInTeam, currentUser, matchTerm) => {
+        // FIXME: We need to include the currentUser here as is not in profilesInTeam on the redux store
```

Comment:
Do we have another  ticket for this?

Reply:
https://mattermost.atlassian.net/browse/RN-385